src/frontend/js/pages/DashboardOrderApplication/index.spec.tsx

```diff
@@ -0,0 +1,80 @@
+import { IntlProvider } from 'react-intl';
+import fetchMock from 'fetch-mock';
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+import { RichieContextFactory as mockRichieContextFactory } from 'utils/test/factories/richie';
+import { Order } from 'types/Joanie';
+import { OrderFactory, TargetCourseFactory } from 'utils/test/factories/joanie';
+import { mockProduct } from 'utils/test/mockProduct';
+import { LearnerDashboardPaths } from 'widgets/Dashboard/utils/learnerRouteMessages';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import { SessionProvider } from 'contexts/SessionContext';
+import { DashboardTest } from 'widgets/Dashboard/components/DashboardTest';
+import { expectBreadcrumbsToEqualParts } from 'utils/test/expectBreadcrumbsToEqualParts';
+import { expectUrlMatchLocationDisplayed } from 'utils/test/expectUrlMatchLocationDisplayed';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockRichieContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+    joanie_backend: { endpoint: 'https://joanie.endpoint' },
+  }).generate(),
+}));
+
+describe('<DashboardOrderApplication />', () => {
+  const WrapperWithDashboard = (route: string) => {
+    return (
+      <QueryClientProvider client={createTestQueryClient({ user: true })}>
+        <IntlProvider locale="en">
+          <SessionProvider>
+            <DashboardTest initialRoute={route} />
+          </SessionProvider>
+        </IntlProvider>
+      </QueryClientProvider>
+    );
+  };
+
+  beforeEach(() => {
+    fetchMock.get('https://joanie.endpoint/api/v1.0/addresses/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/credit-cards/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/enrollments/', []);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+    fetchMock.restore();
+  });
+
+  it('renders breadcrumb and back goes to order', async () => {
+    const order: Order = {
+      ...OrderFactory.generate(),
+      target_courses: [TargetCourseFactory.generate()],
+      enrollments: [],
+    };
+    const product = mockProduct(order);
+    fetchMock.get(
+      'https://joanie.endpoint/api/v1.0/orders/',
+      { results: [order], next: null, previous: null, count: null },
+      { overwriteRoutes: true },
+    );
+
+    render(
+      WrapperWithDashboard(LearnerDashboardPaths.ORDER_APPLICATION.replace(':orderId', order.id)),
+    );
+    await expectBreadcrumbsToEqualParts(['Back', 'My courses', product.title, 'Application']);
+    expectUrlMatchLocationDisplayed(
+      LearnerDashboardPaths.ORDER_APPLICATION.replace(':orderId', order.id),
+    );
+
+    // click on back button.
+    const user = userEvent.setup();
+    await act(async () => user.click(screen.getByRole('link', { name: 'Back' })));
```

Comment:
I see that a simple `await userEvent.click()` throw a act warning.
It may work with a await of a element in the new page. I try to found one to try but i didn't.